presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+        return columns;
+    }
+
+    @Override
+    public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        if (!isAMaterializedView(session, viewName)) {
+            return Optional.empty();
+        }
+
+        Optional<Table> matViewOptional = metastore.getTable(new HiveIdentity(session), viewName.getSchemaName(), viewName.getTableName());
+        if (!matViewOptional.isPresent()) {
+            return Optional.empty();
+        }
+        Table matView = matViewOptional.get();
+
+        ConnectorMaterializedViewDefinition definition = decodeMaterializedViewData(matView.getViewOriginalText()
+                .orElseThrow(() -> new PrestoException(HIVE_INVALID_METADATA, "No view original text: " + viewName)));
+
+        String storageTable = matView.getParameters().getOrDefault(STORAGE_TABLE, "");
+        return Optional.of(new ConnectorMaterializedViewDefinition(
+            definition.getOriginalSql(),
+            storageTable,
+            definition.getCatalog(),
+            Optional.of(viewName.getSchemaName()),
+            definition.getColumns(),
+            definition.getComment(),
+            Optional.of(matView.getOwner().replaceAll("@.*", "")),
+            new HashMap<>(matView.getParameters())));
+    }
+
+    public Optional<io.prestosql.plugin.iceberg.TableToken> getTableToken(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+        if (icebergTable.currentSnapshot() == null) {
+            return Optional.empty();
+        }
+        else {
+            return Optional.of(new io.prestosql.plugin.iceberg.TableToken(icebergTable.currentSnapshot().snapshotId()));
+        }
+    }
+
+    public boolean isTableCurrent(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<io.prestosql.plugin.iceberg.TableToken> tableToken)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        Optional<io.prestosql.plugin.iceberg.TableToken> currentToken = getTableToken(session, handle);
+
+        if (!currentToken.isPresent() && tableToken.isPresent() && tableToken.get().getSnapshotId() == -1) {
+            return true;
+        }
+
+        if (tableToken.isPresent() && currentToken.isPresent() &&
+                tableToken.get().getSnapshotId() == currentToken.get().getSnapshotId()) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        Map<String, Optional<io.prestosql.plugin.iceberg.TableToken>> refreshStateMap = getMaterializedViewToken(session, tableHandle);
+        if (refreshStateMap.size() == 0) {
```

Comment:
Use `isEmpty()` for collections instead of `size() == 0`

Reply:
done